bigtable-hbase-1.x-parent/bigtable-hbase-tools/src/main/java/com/google/cloud/bigtable/hbase/tools/HBaseSchemaTranslator.java

```diff
@@ -0,0 +1,414 @@
+      options.outputFilePath = System.getProperty(OUTPUT_FILE_KEY);
+      options.inputFilePath = System.getProperty(INPUT_FILE_KEY);
+      options.zookeeperQuorum = System.getProperty(ZOOKEEPER_QUORUM_KEY);
+      if (System.getProperty(ZOOKEEPER_PORT_KEY) != null) {
+        options.zookeeperPort = Integer.parseInt(System.getProperty(ZOOKEEPER_PORT_KEY));
+      }
+
+      options.tableNameFilter = System.getProperty(TABLE_NAME_FILTER_KEY);
+
+      // Ensure that the options are set properly
+      // TODO It is possible to validate the options without creating the object, but its less
+      // readable. See if we can make it readable and validate before calling the constructor.
+      try {
+        options.validateOptions();
+      } catch (Exception e) {
+        usage(e.getMessage());
+        throw e;
+      }
+
+      return options;
+    }
+  }
+
+  interface SchemaReader {
+
+    ClusterSchemaDefinition readSchema() throws IOException;
+  }
+
+  static class FileBasedSchemaReader implements SchemaReader {
+
+    String schemaFilePath;
+
+    public FileBasedSchemaReader(String schemaFilePath) {
+      this.schemaFilePath = schemaFilePath;
+    }
+
+    @Override
+    public ClusterSchemaDefinition readSchema() throws IOException {
+      return new ObjectMapper().readValue(new File(schemaFilePath), ClusterSchemaDefinition.class);
+    }
+  }
+
+  static class HBaseSchemaReader implements SchemaReader {
+    private final String tableFilterPattern;
+    private final Admin hbaseAdmin;
+
+    public HBaseSchemaReader(String zookeeperQuorum, int zookeeperPort, String tableFilterPattern)
+        throws IOException {
+
+      // If no filter is provided, use `.*` to match all the tables.
+      this.tableFilterPattern = tableFilterPattern == null ? ".*" : tableFilterPattern;
+
+      // Create the HBase admin client.
+      Configuration conf = HBaseConfiguration.create();
+      conf.set(ZOOKEEPER_PORT_KEY, zookeeperPort + "");
+      conf.set(ZOOKEEPER_QUORUM_KEY, zookeeperQuorum);
+      Connection connection = ConnectionFactory.createConnection(conf);
+      this.hbaseAdmin = connection.getAdmin();
+    }
+
+    @VisibleForTesting
+    HBaseSchemaReader(Admin admin, String tableFilterPattern) {
+      this.hbaseAdmin = admin;
+      // If no filter is provided, use `.*` to match all the tables.
+      this.tableFilterPattern = tableFilterPattern == null ? ".*" : tableFilterPattern;
+    }
+
+    private List<HTableDescriptor> getTables() throws IOException {
+
+      System.out.println("Reading tables from " + hbaseAdmin.getClass().getCanonicalName());
+      // Read the table definitions
+      HTableDescriptor[] tables = null;
+      try {
+        // tables = hbaseAdmin.listTables(".*");
+        // TODO reuse the patterns.
+        tables = hbaseAdmin.listTables(tableFilterPattern);
+      } catch (Exception e) {
+        System.out.println(" ERROR in Getting tables from Hbase: " + e.getMessage());
+        throw e;
```

Comment:
logging & throwing is usally a bad idea because things will end up getting printed twice. I would just throw here and catch in your driver